source/common/config/grpc_mux_impl.cc

```diff
@@ -190,17 +192,46 @@ void GrpcMuxImpl::onDiscoveryResponse(
     std::vector<DecodedResourceRef> all_resource_refs;
     OpaqueResourceDecoder& resource_decoder =
         api_state_[type_url].watches_.front()->resource_decoder_;
+
+    auto ttl_expiry_callback = [this, type_url](const auto& expired) {
```

Comment:
Can you add some comments? It's kind of hard to know what some of these almost-plain-English things do.